apsuite/commissioning_scripts/meas_coupling_tune.py

```diff
@@ -17,201 +41,203 @@ class FitTunes(_SimulAnneal):
+        ini_param = self._calc_init_parms(qcurr, tune1, tune2)
+        # least squares using Levenberg-Marquardt minimization algorithm
+        fit_param = least_squares(
+            fun=MeasCoupling._err_func, x0=ini_param,
+            args=(qcurr, tune1, tune2), method='lm')
+
+        self.analysis['qcurr'] = qcurr
+        self.analysis['tune1'] = tune1
+        self.analysis['tune2'] = tune2
+        self.analysis['initial_param'] = ini_param
+        self.analysis['fitted_param'] = fit_param
+        fit_error = self._calc_fitting_error()
+        self.analysis['fitting_error'] = fit_error
+
+    def plot_fitting(
+            self, oversampling=1, title=None, xlabel=None, fig=None,
+            save=False, fname=None):
+        """."""
+        anl = self.analysis
+        fit_vec = anl['fitted_param']['x']
+        qcurr, tune1, tune2 = anl['qcurr'], anl['tune1'], anl['tune2']
+
+        qcurr_interp = _np.interp(
+            _np.arange(0, len(qcurr), 1/oversampling),
+            _np.arange(0, len(qcurr)), qcurr)
+
+        fittune1, fittune2 = MeasCoupling._get_normal_modes(
+            params=fit_vec, curr=qcurr_interp)
 
+        if fig is None:
+            fig = _plt.figure(figsize=(8, 6))
         if xlabel is None:
             xlabel = 'Quadrupole Integrated Strength [1/m]'
         if title is None:
-            title = 'Transverse Linear Coupling: {:.2f} ± {:.2f}%'.format(
-                position[-1] * 100, coup_error * 100)
-
-        prop_cycle = _plt.rcParams['axes.prop_cycle']
-        colors = prop_cycle.by_key()['color']
-        if fig is None:
-            fig = _plt.figure()
-        param, fittune1, fittune2 = FitTunes.calc_tunes(param, *position)
+            title = 'Transverse Linear Coupling: ({:.2f} ± {:.2f}) %'.format(
+                fit_vec[-1]*100, anl['fitting_error'][-1] * 100)
 
         # plot meas data
-        _plt.plot(
-            self._param, self._tune1, 'o', color=colors[0],
-            label='measurement')
-        _plt.plot(self._param, self._tune2, 'o', color=colors[0])
+        _plt.plot(qcurr, tune1, 'o', color='C0', label=r'$\nu_1$')
+        _plt.plot(qcurr, tune2, 'o', color='C1', label=r'$\nu_2$')
 
         # plot fitting
-        _plt.plot(param, fittune1, color=colors[1], label='fitting')
-        _plt.plot(param, fittune2, color=colors[1])
+        _plt.plot(qcurr_interp, fittune1, color='tab:gray', label='fitting')
+        _plt.plot(qcurr_interp, fittune2, color='tab:gray')
         _plt.legend()
         _plt.xlabel(xlabel)
         _plt.ylabel('Transverse Tunes')
         _plt.title(title)
-        _plt.grid()
+        _plt.grid(ls='--', alpha=0.5)
+        _plt.tight_layout(True)
+        if save:
+            if fname is None:
+                date_string = _time.strftime("%Y-%m-%d-%H:%M")
+                fname = 'coupling_fitting_{}.png'.format(date_string)
+            _plt.savefig(fname, format='png', dpi=300)
         _plt.show()
```

Comment:
I think it is a good practice to return the `Figure` and the `Axes` objects for the user to post configure the graph.
I think it is better not to call `_plt.show()` either.